### PR TITLE
Implemented logic to make selectProduct page functional(#29pt6c6)

### PR DESCRIPTION
**Pull request by disha1202**
Screenshot from 2022-05-04 17-24-51

### CODE REVIEW

File: src/store/modules/product/actions.ts [hunk 1]
```diff
@@ -46,6 +46,89 @@ const actions: ActionTree<ProductState, RootState> = {
...
+        products = products.map((product: any) => {
+          return {
+            productId: product.groupValue,
+            productName: product.doclist.docs[0]?.parentProductName,
+            variants: product.doclist.docs
+          }
+        })
+
+        let productIds: any = new Set();
+        products.forEach((product: any) => {
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts:99]: Use map in place of forEach everywhere.

File: src/store/modules/product/actions.ts [hunk 2]
```diff
@@ -46,6 +46,89 @@ const actions: ActionTree<ProductState, RootState> = {
...
+          if(product.productId) productIds.add(product.productId);
+        })
+        productIds = [...productIds]
+        const productInformation = await this.dispatch("product/fetchProducts", { productIds });
+
+        products = products.map((product: any) => {
+          product.variants.forEach((variant: any) => {
+            variant.isSelected = false;
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts:107]: I think we don't need this logic as we are not having functionality to select a single product.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/product/actions.ts (hunk 1 above)
Comment (disha1202) [reply src/store/modules/product/actions.ts]: Replaced forEach with map

File: src/store/modules/product/actions.ts (hunk 2 above)
Comment (disha1202) [reply src/store/modules/product/actions.ts]: Removed

File: src/store/modules/product/actions.ts [hunk 3]
```diff
@@ -46,6 +45,36 @@ const actions: ActionTree<ProductState, RootState> = {
...
+            productId: product.groupValue,
+            productName: product.doclist.docs[0]?.parentProductName,
+            variants: product.doclist.docs
+          }
+        })
+
+        if(payload.json.params.start && payload.json.params.start > 0) products = state.products.list.concat(products);
+        commit(types.PRODUCT_LIST_UPDATED, { products, virtualCount, variantCount });
+      } else {
+        showToast(translate("Products not found"));
```
Comment (ymaheshwari1) [inline src/store/modules/product/actions.ts:69]: Add this to en.json

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/product/actions.ts (hunk 3 above)
Comment (ymaheshwari1) [reply src/store/modules/product/actions.ts:69]: @meet-aniket

Comment (meet-aniket) [reply src/store/modules/product/actions.ts:69]: Okay, Sir.

Comment (meet-aniket) [reply src/store/modules/product/actions.ts:69]: Added this label on ```en.json``` file.

**Review by ymaheshwari1 — APPROVED** [review]

File: src/store/modules/product/actions.ts [hunk 4]
```diff
@@ -32,7 +32,6 @@ const actions: ActionTree<ProductState, RootState> = {
         const totalProductsCount = resp.data.response.numFound;
 
         if (payload.viewIndex && payload.viewIndex > 0) products = state.products.list.concat(products)
-        commit(types.PRODUCT_SEARCH_UPDATED, { products: products, totalProductsCount: totalProductsCount })
```
Comment (adityasharma7) [inline src/store/modules/product/actions.ts:35]: If find product action is not used anywhere, we could remove it

File: src/store/modules/product/ProductState.ts [hunk 5]
```diff
@@ -1,7 +1,11 @@
 export default interface ProductState {
+  cached: any;
```
Comment (adityasharma7) [inline src/store/modules/product/ProductState.ts:2]: We could revert changes done for cached products if not in use

File: src/store/modules/product/actions.ts [hunk 6]
```diff
@@ -46,6 +45,36 @@ const actions: ActionTree<ProductState, RootState> = {
...
 
+  async getProducts({ commit, state }, payload) {
+    let resp;
+
+    try {
+      resp = await ProductService.getProducts(payload);
+
+      if (resp.status === 200 && resp.data.grouped.groupId?.ngroups > 0 && !hasError(resp)) {
+        let products = resp.data.grouped.groupId?.groups;
+        const virtualCount = resp.data.grouped.groupId.ngroups;
```
Comment (adityasharma7) [inline src/store/modules/product/actions.ts:56]:
```suggestion
        const totalVirtual = resp.data.grouped.groupId.ngroups;
```

File: src/store/modules/product/actions.ts [hunk 7]
```diff
@@ -46,6 +45,36 @@ const actions: ActionTree<ProductState, RootState> = {
...
+        const variantCount = resp.data.grouped.groupId.matches;
```
Comment (adityasharma7) [inline src/store/modules/product/actions.ts:57]:
```suggestion
        const totalVariant = resp.data.grouped.groupId.matches;
```

File: src/views/SelectProduct.vue [hunk 8]
```diff
@@ -112,42 +112,48 @@
...
+            </section>
+  
+            <section class="section-grid">
+              <div v-for="variant in product.variants" :key="variant">
+                <ion-card>
+                  <Image :src="variant.mainImageUrl" />
+                  <ion-item lines="none">
+                    <ion-label>
+                      {{ variant.productName }}
+                      <p>{{ $t("Color") }}: {{ $filters.getFeature(variant.featureHierarchy, '1/COLOR/') }}</p>
```
Comment (adityasharma7) [inline src/views/SelectProduct.vue:145]: I think we should extract it while getting products

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/product/actions.ts (hunk 4 above)
Comment (meet-aniket) [reply src/store/modules/product/actions.ts:35]: Removed `findProduct` action from the product module.

File: src/store/modules/product/ProductState.ts (hunk 5 above)
Comment (meet-aniket) [reply src/store/modules/product/ProductState.ts:2]: Reverted changes related to cached products.

File: src/store/modules/product/actions.ts (hunk 6 above)
Comment (meet-aniket) [reply src/store/modules/product/actions.ts:21]: Updated variable name.

File: src/store/modules/product/actions.ts (hunk 7 above)
Comment (meet-aniket) [reply src/store/modules/product/actions.ts:22]: Updated variable name.

File: src/views/SelectProduct.vue (hunk 8 above)
Comment (meet-aniket) [reply src/views/SelectProduct.vue]: Updated code to extract features related to each variant at the time of API call and directly use it on UI.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**